positron/electron/lib/renderer/web-view/web-view.js

```diff
@@ -264,10 +269,92 @@ var WebViewImpl = (function() {
-    this.setAttribute('type', 'application/browser-plugin');
-    this.setAttribute('id', 'browser-plugin-' + getNextId());
+    this.setAttribute('mozbrowser', 'true');
+
+    // This breaks loading a page in the mozbrowser when we give the app page
+    // the system principal, so we've temporarily disabled it.
+    // this.setAttribute('remote', 'true');
+
+    this.addEventListener('mozbrowserloadstart', function(event) {
+      var internal;
+      internal = v8Util.getHiddenValue(this, 'internal');
+      if (!internal) {
+        return;
+      }
+
+      ipcRenderer.emit("ATOM_SHELL_GUEST_VIEW_INTERNAL_DISPATCH_EVENT-" + internal.viewInstanceId,
+                       event, 'did-start-loading');
+    }, false);
+
+    // https://wiki.mozilla.org/WebAPI/BrowserAPI/Common_Subset#DOM_Events
+    // suggests that Google's load-commit is the equivalent of Mozilla's
+    // mozbrowserloadend, while Mozilla's mozbrowserlocationchange doesn't have
+    // a Google equivalent.  But I suspect that load-commit is actually
+    // similar to mozbrowserlocationchange, at least for top-frame loads.
+    //
+    // However, load-commit also fires on sub-frame loads, which isn't the case
+    // for mozbrowserlocationchange.  This implementation also doesn't yet set
+    // the url and isMainFrame properties.
+    //
+    // TODO: complete support for the load-commit event.
+    // https://github.com/mozilla/positron/issues/70
+    //
+    this.addEventListener('mozbrowserlocationchange', function(event) {
+      var internal;
+      internal = v8Util.getHiddenValue(this, 'internal');
+      if (!internal) {
+        return;
+      }
+
+      ipcRenderer.emit("ATOM_SHELL_GUEST_VIEW_INTERNAL_DISPATCH_EVENT-" + internal.viewInstanceId,
+                       event, 'load-commit');
+    }, false);
+
+    this.addEventListener('mozbrowserloadend', function(event) {
+      var internal;
+      internal = v8Util.getHiddenValue(this, 'internal');
+      if (!internal) {
+        return;
+      }
+
+      ipcRenderer.emit("ATOM_SHELL_GUEST_VIEW_INTERNAL_DISPATCH_EVENT-" + internal.viewInstanceId,
+                       event, 'did-stop-loading');
+      ipcRenderer.emit("ATOM_SHELL_GUEST_VIEW_INTERNAL_DISPATCH_EVENT-" + internal.viewInstanceId,
+                       event, 'did-finish-load');
+    }, false);
+
+    this.addEventListener('mozbrowsererror', function(event) {
+      var internal;
+      internal = v8Util.getHiddenValue(this, 'internal');
+      if (!internal) {
+        return;
+      }
+
+      ipcRenderer.emit("ATOM_SHELL_GUEST_VIEW_INTERNAL_DISPATCH_EVENT-" + internal.viewInstanceId,
+                       event, 'did-stop-loading');
+      // TODO: support the 'did-fail-load' event properties.
+      ipcRenderer.emit("ATOM_SHELL_GUEST_VIEW_INTERNAL_DISPATCH_EVENT-" + internal.viewInstanceId,
+                       event, 'did-fail-load');
+    }, false);
+
+    // this.setAttribute('type', 'application/browser-plugin');
```

Comment:
> Why is this commented out?

The _type_ attribute is specific to _object_ elements and isn't defined for _iframe_ elements. I added a comment to that effect and also updated other comments and rearranged code to make it the changes clearer in 688b355.